packages/gatsby-plugin-manifest/src/gatsby-node.js

```diff
@@ -76,6 +76,10 @@ exports.onPostBootstrap = async ({ reporter }, { localize, ...manifest }) => {
         makeManifest(cache, reporter, {
           ...manifest,
           ...locale,
+          ...{ cache_busting_mode: `name` },
+          /* localization requires unique filenames for output files if a different src Icon is defined.
```

Comment:
@moonmeister I'm looking into modifying this plugin to be more extensible and fix some existing bugs like #17255. I'm trying to understand some of the cache busting logic but parts of it are pretty confusing so I was hoping you could explain some parts. Why does localization require name-based cache busting rather than being query-based?

Reply:
@dkthehuman first the code you show above is outdated, there are more recent commits that only require name based busting when you override the icon for a specific locale.  Without this the icons of the different locales would overwrite each other.  When I added caching to the icon generation i was able to detect if it was a different icon being generated for different locales, but I still needed away to write a different file name. The name based cache busting does just that, so it's required. FYI: cache busting doesn't generate a random hash to bust the cache. It generates a hash based on the source image. meaning you can run the build infinite times and cache busting will always have the same hash in the name or query string.

Reply:
Ah, gotcha. Thanks!